### PR TITLE
Feat/location page

### PR DESCRIPTION
**Pull request by klhq**
## Purpose

* Refactor actions, reducer, & types of ```labelingPage```
* Location page
* Location register page
* TopNav
* Home page
* Modify LabelingPage layout & add labelingType config
* Add Breadcrumb

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone git@github.com:linkernetworks/azure-intelligent-edge-patterns.git
git checkout feat/locationPage
cd VisonOnEdge/ui

yarn
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: VisionOnEdge/ui/src/components/MainLayout/MainLayout.tsx [hunk 1]
```diff
@@ -38,7 +24,51 @@ export const MainLayout: React.FC = ({ children }) => {
...
+//       boxShadow: '0px 1px 10px 0px rgba(0,0,0,0.75)',
+//       zIndex: 2,
+//       padding: 0,
+//       fontSize: '20px',
+//       display: 'flex',
+//       alignItems: 'center',
+//       paddingLeft: '1em',
+//     }}
+//   />
+// );
```
Comment (andykao1213) [inline VisionOnEdge/ui/src/components/MainLayout/MainLayout.tsx]: Should this be removed?

File: VisionOnEdge/ui/src/components/ImageLink.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  defaultSrc,
+  label = '',
+  imgPadding = '10px',
+  width = '100px',
+  height = '100px',
+}) => {
+  return (
+    <Flex column styles={{ width }}>
+      <Link to={to}>
+        <FlexItem styles={{ height }}>
```
Comment (andykao1213) [inline VisionOnEdge/ui/src/components/ImageLink.tsx]: The images ratios looks wrong by using this method, I suggest the method below:
```
        <div style={{ height, width }}>
          <Image
            src={imgSrc ?? defaultSrc}
            styles={{ padding: imgPadding, width: '100%', height: '100%', objectFit: 'contain' }}
          />
        </div>
```

File: VisionOnEdge/ui/src/pages/LocationRegister.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        </FlexItem>
+        <Input
+          styles={{ paddingTop: '10px', height: '100px' }}
+          placeholder="Location Name"
+          value={locationRegisterInput.name}
+          onChange={(_, newProps): void =>
+            setLocationRegisterInput((prev) => ({ ...prev, name: newProps.value }))
+          }
+        />
+      </Flex>
```
Comment (andykao1213) [inline VisionOnEdge/ui/src/pages/LocationRegister.tsx]: The Input doesn't align with the input below. Try using Grid as below?

**Review by andykao1213 — CHANGES_REQUESTED** [review]

File: VisionOnEdge/ui/src/components/ImageLink.tsx (hunk 2 above)
Comment (klhq) [reply VisionOnEdge/ui/src/components/ImageLink.tsx]: It looks better now. Thanks for the advice.

**Review by andykao1213 — APPROVED** [review]

**Pull request merged**